tutorials/text/text_intro.py

```diff
@@ -93,3 +96,337 @@
+# labels fit in the right-hand plot.
+
+fig, axs = plt.subplots(2, 2, figsize=(8, 5), tight_layout=True)
+axs = axs.flatten()
+for n, ax in enumerate(axs):
+    ax.plot(x1*10., y1)
+
+formatter = matplotlib.ticker.FormatStrFormatter('%1.1f')
+locator = matplotlib.ticker.MaxNLocator(nbins='auto', steps=[1, 4, 10])
+axs[1].xaxis.set_major_locator(locator)
+axs[1].xaxis.set_major_formatter(formatter)
+
+formatter = matplotlib.ticker.FormatStrFormatter('%1.5f')
+locator = matplotlib.ticker.AutoLocator()
+axs[2].xaxis.set_major_formatter(formatter)
+axs[2].xaxis.set_major_locator(locator)
+
+formatter = matplotlib.ticker.FormatStrFormatter('%1.5f')
+locator = matplotlib.ticker.MaxNLocator(nbins=4)
+axs[3].xaxis.set_major_formatter(formatter)
+axs[3].xaxis.set_major_locator(locator)
+
+plt.show()
+
+##############################################################################
+#  Finally, we can specify functions for the formatter using
+# `matplotlib.ticker.FuncFormatter`.
+
+
+def formatoddticks(x, pos):
+    """Format odd tick positions
+    """
+    if x % 2:
+        return '%1.2f' % x
+    else:
+        return ''
+
+fig, ax = plt.subplots(figsize=(5, 3), tight_layout=True)
+ax.plot(x1, y1)
+formatter = matplotlib.ticker.FuncFormatter(formatoddticks)
+locator = matplotlib.ticker.MaxNLocator(nbins=6)
+ax.xaxis.set_major_formatter(formatter)
+ax.xaxis.set_major_locator(locator)
+
+plt.show()
+
+##############################################################################
+# Dateticks
+# ~~~~~~~~~
+#
+# Matplotlib can accept `datetime.datetime` and `numpy.datetime64`
+# objects as plotting arguments.  Dates and times require special
+# formatting, which can often benefit from manual intervention. In
+# order to help, dates have spectial Locators and Formatters,
+# defined in the `matplotlib.dates` module.
+#
+# A simple exmample is as follows.  Note how we have to rotate the
+# tick labels so that they don't over-run each other.
+import datetime
+
+fig, ax = plt.subplots(figsize=(5, 3), tight_layout=True)
+base = datetime.datetime(2017, 1, 1, 0, 0, 1)
+time = [base + datetime.timedelta(days=x) for x in range(len(y1))]
+
+ax.plot(time, y1)
+ax.tick_params(axis='x', rotation=70)
+plt.show()
+
+##############################################################################
+# Maybe the format of the labels above is acceptable, but the choices is
+# rather idiosyncratic.  We can make the ticks fall on the start of the month
+# by very lightly modifying `matplotlib.dates.AutoDateLocator`
+import matplotlib.dates as mdates
+
+locator = mdates.AutoDateLocator(interval_multiples=True)
+
+fig, ax = plt.subplots(figsize=(5, 3), tight_layout=True)
+ax.xaxis.set_major_locator(locator)
+ax.plot(time, y1)
+ax.tick_params(axis='x', rotation=70)
+plt.show()
+
+##############################################################################
+# Oddly, this changes the labels!  The easiest fix is to pass a format to
```

Comment:
While the behavior may be "odd" I think we should not say it as such in the official docs... (a brief explaination why (I guess locators and formatters are coupled) would be nicer)

Reply:
Yeah, I'm not sure why.  Seems like a bug.  But for now its a good excuse to introduce the formatter...

Reply:
struck inflamatory language for:  `However, this changes the tick labels.`